docs/CONTRIBUTING.md

```diff
@@ -19,33 +19,130 @@ Please submit pull requests in response to open issues. If you have a bug or fea
-- Redis for Windows - You can use [this](https://github.com/tporadowski/redis/releases) installer to get up and running quickly for now.
-- Redis for MacOS - need instructions
-  An easier solution would be to use Docker.
+#### Linux:
 
-**Setup**
+Install Redis using your distribution's package manager, for example:
+
+- Ubuntu based: `sudo apt install redis`
+- Red Hat, Fedora: `sudo dnf install redis`
+
+_Once Redis is installed, you can start it in a terminal by running:_
+
+```
+redis-server
+```
+
+#### Windows:
+
+We are using [Chocolatey package manager](https://chocolatey.org/) to install Redis on Windows. To get Chocolatey, simply follow this [guide](https://chocolatey.org/install) and run the following commands:
+
+1. To install Redis: `choco install redis-64 -v`
+1. To set Redis as a windows service: `redis-server --service-install`
+1. To start Redis: `redis-server --service-start`
+1. To check if running and display server information: `redis-cli info`
+
+### Docker and Docker-Compose Set Up
+
+#### Windows 10 Pro, Enterprise, or Education
+
+1. Get [Docker for Desktop For Windows](https://hub.docker.com/editions/community/docker-ce-desktop-windows)
+1. Docker for Desktop comes with docker-compose installed.
+
+#### MacOS (Sierra 10.12 or above)
+
+1. Get [Docker for Desktop For Mac](https://hub.docker.com/editions/community/docker-ce-desktop-mac)
+1. Docker for Desktop comes with docker-compose installed.
+
+#### Windows 10 Home Edition and Windows Subsystem for Linux (WSL)
+
+Docker Desktop for Windows is not available on Home Edition, and you cannot run docker in WSL (Windows Subsystem for Linux). You can get the environment set up using **one** of these three methods:
+
+- Get [Docker Toolbox](https://docs.docker.com/toolbox/toolbox_install_windows/) instead of Docker Desktop for Windows. Make sure that your system can do virtualization and enable virtualization.
+- Set up a virtual machine to run Linux Ubuntu
+  1. Make sure your system can do virtualization and enable it.
+  1. [Download VirtualBox](https://www.virtualbox.org/wiki/Downloads)
+  1. [Follow this helpful youtube tutorial to create a virtual machine with Ubuntu](https://www.youtube.com/watch?v=ThsxqznrgCw&t=401s)
+  1. Use the Linux installation instructions [below](#linux-Ubuntu).
+- Update your home edition to Windows 10 Education Edition
+  1. Download Windows 10 Education Edition from the [Seneca Software Center](https://senecacollege.onthehub.com/WebStore/OfferingDetails.aspx?o=c0bd2c36-a530-e511-940e-b8ca3a5db7a1)
+  1. Update your OS using the installation instructions.
+  1. Use [Windows 10 Education Edition](#Windows-10-Pro,-Enterprise,-or-Education) set up instructions.
+
+#### Linux-Ubuntu
+
+This guide is sourced from the official [Docker-CE](https://docs.docker.com/install/linux/docker-ce/ubuntu/) and [Docker-Compose](https://docs.docker.com/compose/install/) Installation Documentation.
+
+**Install Docker Engine - Community Edition**
```

Comment:
I think you're missing adding your user to the docker's group.
Instructions for that [here](https://docs.docker.com/install/linux/linux-postinstall/).
Also, there are instructions to run docker as a service, in case you wanted to add that as well. It might be useful.

Reply:
Thanks for the feedback, I will make these changes and get back to you!